utils/utils.go

```diff
@@ -662,7 +669,13 @@ func CustomizeServiceAccount(sa *corev1.ServiceAccount, ba common.BaseComponent)
 	sa.Labels = ba.GetLabels()
 	sa.Annotations = MergeMaps(sa.Annotations, ba.GetAnnotations())
 
+	psr := ba.GetStatus().GetReferences()[common.StatusReferencePullSecretName]
+	if psr != "" && (ba.GetPullSecret() == nil || *ba.GetPullSecret() != psr) {
+		removePullSecret(sa, psr)
+	}
+
 	if ba.GetPullSecret() != nil {
+		ba.GetStatus().SetReference(common.StatusReferencePullSecretName, *ba.GetPullSecret())
 		if len(sa.ImagePullSecrets) == 0 {
```

Comment:
Adding the pull secret by iterating through the secrets would be better than adding it based on index. WSA for [reference](https://github.ibm.com/websphere/automation-operator/blob/ab96861893c32f30f9918ee8d939db7b7c6fcaa6/controllers/websphereautomation_controller.go#L543)